src/test/kotlin/uk/gov/justice/digital/hmpps/sentenceplan/integration/GoalControllerTest.kt

```diff
@@ -250,4 +250,22 @@ class GoalControllerTest : IntegrationTestBase() {
       .exchange()
       .expectStatus().isCreated
   }
+
+  @Test
+  fun `delete goal should return no content`() {
+    webTestClient.delete().uri("/goals/ede47f7f-8431-4ff9-80ec-2dd3a8db3841")
+      .header("Content-Type", "application/json")
+      .headers(setAuthorisation(user = "Tom C", roles = listOf("ROLE_RISK_INTEGRATIONS_RO")))
+      .exchange()
+      .expectStatus().isNoContent
+  }
+
+  @Test
+  fun `deleting a goal that does not exist should return 404`() {
+    webTestClient.delete().uri("/goals/93ab5028-867f-4554-aa5a-2383e6b50f1f")
+      .header("Content-Type", "application/json")
+      .headers(setAuthorisation(user = "Tom C", roles = listOf("ROLE_RISK_INTEGRATIONS_RO")))
+      .exchange()
+      .expectStatus().isNotFound
```

Comment:
Could we also get a test which proves that the steps related to a goal are deleted when the goal is deleted?

Reply:
Additional calls added to show goal and steps are deleted.